.github/workflows/release.yml

```diff
@@ -0,0 +1,45 @@
+name: Publish releases
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  build-and-publish-pypi:
+    name: Builds and publishes releases to PyPI
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.vars.outputs.tag }}
+    steps:
+      - uses: actions/checkout@v4.0.0
+      - name: Get tag
+        id: vars
+        run: echo "tag=${GITHUB_REF#refs/*/}" >> $GITHUB_OUTPUT
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4.7.0
+        with:
+          python-version: "3.10"
+      - name: Install build
+        run: >-
+          pip install build tomli tomli-w
+      - name: Set Python project version from tag
+        shell: python
+        run: |-
+          import tomli
+          import tomli_w
+
+          with open("pyproject.toml", "rb") as f:
+            pyproject = tomli.load(f)
+
+          pyproject["project"]["version"] = "${{ steps.vars.outputs.tag }}"
+
+          with open("pyproject.toml", "wb") as f:
+            tomli_w.dump(pyproject, f)
+      - name: Build
+        run: >-
+          python3 -m build
+      - name: Publish release to PyPI
+        uses: pypa/gh-action-pypi-publish@v1.8.10
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_TOKEN }}
```

Comment:
We could directly switch to [Trusted Publisher](https://docs.pypi.org/trusted-publishers/) to improve security

Reply:
Ah yes, can we do that in a follow up? I'm not sure how to do that with the libs org and I think the PYPI account linked is still from @balloob